core/src/main/java/org/infinispan/interceptors/distribution/BaseDistributionInterceptor.java

```diff
@@ -1037,4 +1036,10 @@ public Object visitRemoveExpiredCommand(InvocationContext ctx, RemoveExpiredComm
          }
       }
    }
+
+   public Object maxIdleRemoval(InvocationContext ctx, RemoveExpiredCommand command, DistributionInfo distributionInfo)
```

Comment:
Could be `protected`, and I'd prefer a more "active" name, say `invokeRemoveExpiredCommand`

Looking at `NonTxDistributionInterceptor.handleNonTxWriteCommand()` it also checks `ctx.isOriginLocal()`, so I wonder if we need some tests for clustered expiration in a non-tx cache that uses it (with `PrivateGlobalConfigurationBuilder.serverMode()`).

Reply:
Sure

Let me look at it closer.

Reply:
As far as I can tell, the `NonTxDistributionInterceptor.handleNonTxWriteCommand` is only invoked for a `RemoveExpiredCommand` with lifespan, in which case this isn't an issue.

Also this issue was because `TriangleDistributionInterceptor` has the backup node send the response to a different node then the message was sent from.

I will add a test to the server ones though.

Reply:
Added a test to `MixedExpiryTest` for this.